File: jetbrains/src/main/kotlin/identitybraces/core/TraitIds.kt
package identitybraces.core

/**
 * Every trait's stable identifier.
 *
 * These strings are the persistence key — they appear in the settings as `wizard=4`, and in
 * the Visual Studio extension's settings.ini as `Trait.wizard=4` — so renaming one silently
 * resets that trait's weight for everyone who had customised it. Add freely; rename never.
 */
object TraitIds {
    // ---- Body: what the glyph is. One per brace. ----
    const val QUESTION = "question"
    const val UNICODE_VARIANT = "unicode"
    const val WRONG_BRACKET = "wrongbracket"
    const val MIRRORED = "mirrored"
    const val UPSIDE_DOWN = "upsidedown"
    const val EMOJI = "emoji"
    const val FOREIGN_FONT = "foreignfont"
    const val SUBSCRIPT = "subscript"

    // ---- Creature: ears, horns, silhouettes. One per brace. ----
    const val CATGIRL = "catgirl"
    const val BUNNY = "bunny"
    const val DEVIL = "devil"
    const val ANGEL = "angel"
    const val GHOST = "ghost"
    const val WIZARD = "wizard"
    const val VAMPIRE = "vampire"
    const val BEE = "bee"
    const val FROG = "frog"
    const val SNAKE = "snake"
    const val CRAB = "crab"
    const val BAT = "bat"
    const val SPIDER = "spider"
    const val FOX = "fox"
    const val WOLF = "wolf"
    const val DRAGON = "dragon"
    const val UNICORN = "unicorn"
    const val PENGUIN = "penguin"
    const val OWL = "owl"
    const val CTHULHU = "cthulhu"
    const val SLIME = "slime"
    const val MUSHROOM = "mushroom"
    const val CACTUS = "cactus"
    const val ROBOT = "robot"
    const val PIRATE = "pirate"

    // ---- Costume: worn over the glyph. One per brace. ----
    const val THIGH_HIGHS = "thighhighs"
    const val TOP_HAT = "tophat"
    const val CROWN = "crown"
    const val SUNGLASSES = "sunglasses"
    const val SCARF = "scarf"
    const val BOWTIE = "bowtie"
    const val CAPE = "cape"
    const val PARTY_HAT = "partyhat"
    const val HEADPHONES = "headphones"
    const val FLOWER_CROWN = "flowercrown"
    const val BEANIE = "beanie"
    const val MONOCLE = "monocle"
    const val NECKTIE = "necktie"
    const val BACKPACK = "backpack"
    const val WINGS = "wings"
    const val ARMOUR = "armour"
    const val BANDAGE = "bandage"
    const val MOUSTACHE = "moustache"

    // ---- Motion: one per brace, or they fight over the transform. ----
    const val COLOUR_CYCLE = "cycle"
    const val WOBBLE = "wobble"
    const val BOUNCE = "bounce"
    const val BREATHE = "breathe"
    const val HEARTBEAT = "heartbeat"
    const val SHIVER = "shiver"
    const val BLINK = "blink"
    const val SPIN = "spin"
    const val FLIP = "flip"
    const val GLITCH = "glitch"
    const val SPARKLE = "sparkle"
    const val DRIFT = "drift"
    const val SHIMMER = "shimmer"
    const val FLICKER = "flicker"
    const val WAVE = "wave"
    const val TYPEWRITER = "typewriter"

    // ---- Effect: any number, rolled independently. ----
    const val FIRE = "fire"
    const val GRADIENT_FILL = "gradient"
    const val BOLD_ITALIC = "bolditalic"
    const val TILTED = "tilted"
    const val UNDERLINE = "underline"
    const val SHADOW = "shadow"

    /**
     * Sweating and unsteady. Rollable like any other effect, but also forced on by the
     * complexity warning, which is the only trait in the catalogue with a predicate behind
     * it as well as a weight.
     */
    const val DISTRESSED = "distressed"

    // ---- Chaos: behaviours that react to the world rather than sit still. ----
    const val FLEE_CURSOR = "fleecursor"
    const val NAMED = "named"
    const val NOCTURNAL = "nocturnal"
    const val SEASONAL = "seasonal"
    const val BUILD_REACTIVE = "buildreactive"
    const val SWAP_PLACES = "swapplaces"
    const val DRUNK = "drunk"
    const val GRAVITY = "gravity"
    const val STAGE_FRIGHT = "stagefright"
    const val MITOSIS = "mitosis"
    const val TABLE_FLIP = "tableflip"
    const val FIRE_BRIGADE = "firebrigade"
}

File: src/IdentityBraces/Core/TraitIds.cs
namespace IdentityBraces.Core
{
    /// <summary>
    /// Every trait's stable identifier.
    /// </summary>
    /// <remarks>
    /// These strings are the persistence key — they appear in settings.ini as
    /// <c>Trait.wizard=4</c> — so renaming one silently resets that trait's weight for
    /// everyone who had customised it. Add freely; rename never.
    /// </remarks>
    internal static class TraitIds
    {
        // ---- Body: what the glyph is. One per brace. ----
        public const string Question = "question";
        public const string UnicodeVariant = "unicode";
        public const string WrongBracket = "wrongbracket";
        public const string Mirrored = "mirrored";
        public const string UpsideDown = "upsidedown";
        public const string Emoji = "emoji";
        public const string ForeignFont = "foreignfont";
        public const string Subscript = "subscript";

        // ---- Creature: ears, horns, silhouettes. One per brace. ----
        public const string Catgirl = "catgirl";
        public const string Bunny = "bunny";
        public const string Devil = "devil";
        public const string Angel = "angel";
        public const string Ghost = "ghost";
        public const string Wizard = "wizard";
        public const string Vampire = "vampire";
        public const string Bee = "bee";
        public const string Frog = "frog";
        public const string Snake = "snake";
        public const string Crab = "crab";
        public const string Bat = "bat";
        public const string Spider = "spider";
        public const string Fox = "fox";
        public const string Wolf = "wolf";
        public const string Dragon = "dragon";
        public const string Unicorn = "unicorn";
        public const string Penguin = "penguin";
        public const string Owl = "owl";
        public const string Cthulhu = "cthulhu";
        public const string Slime = "slime";
        public const string Mushroom = "mushroom";
        public const string Cactus = "cactus";
        public const string Robot = "robot";
        public const string Pirate = "pirate";

        // ---- Costume: worn over the glyph. One per brace. ----
        public const string ThighHighs = "thighhighs";
        public const string TopHat = "tophat";
        public const string Crown = "crown";
        public const string Sunglasses = "sunglasses";
        public const string Scarf = "scarf";
        public const string Bowtie = "bowtie";
        public const string Cape = "cape";
        public const string PartyHat = "partyhat";
        public const string Headphones = "headphones";
        public const string FlowerCrown = "flowercrown";
        public const string Beanie = "beanie";
        public const string Monocle = "monocle";
        public const string Necktie = "necktie";
        public const string Backpack = "backpack";
        public const string Wings = "wings";
        public const string Armour = "armour";
        public const string Bandage = "bandage";
        public const string Moustache = "moustache";

        // ---- Motion: one per brace, or they fight over the transform. ----
        public const string ColourCycle = "cycle";
        public const string Wobble = "wobble";
        public const string Bounce = "bounce";
        public const string Breathe = "breathe";
        public const string Heartbeat = "heartbeat";
        public const string Shiver = "shiver";
        public const string Blink = "blink";
        public const string Spin = "spin";
        public const string Flip = "flip";
        public const string Glitch = "glitch";
        public const string Sparkle = "sparkle";
        public const string Drift = "drift";
        public const string Shimmer = "shimmer";
        public const string Flicker = "flicker";
        public const string Wave = "wave";
        public const string Typewriter = "typewriter";

        // ---- Effect: any number, rolled independently. ----
        public const string Fire = "fire";
        public const string GradientFill = "gradient";
        public const string BoldItalic = "bolditalic";
        public const string Tilted = "tilted";
        public const string Underline = "underline";
        public const string Shadow = "shadow";

        /// <summary>
        /// Sweating and unsteady. Rollable like any other effect, but also forced on by the
        /// complexity warning, which is the only trait in the catalogue with a predicate
        /// behind it as well as a weight.
        /// </summary>
        public const string Distressed = "distressed";

        // ---- Chaos: behaviours that react to the world rather than sit still. ----
        public const string FleeCursor = "fleecursor";
        public const string Named = "named";
        public const string Nocturnal = "nocturnal";
        public const string Seasonal = "seasonal";
        public const string BuildReactive = "buildreactive";
        public const string SwapPlaces = "swapplaces";
        public const string Drunk = "drunk";
        public const string Gravity = "gravity";
        public const string StageFright = "stagefright";
        public const string Mitosis = "mitosis";
        public const string TableFlip = "tableflip";
        public const string FireBrigade = "firebrigade";
    }
}
